packages/app/src/specs/SpecsListBanners.cy.tsx

```diff
@@ -48,6 +55,75 @@ describe('<SpecsListBanners />', () => {
     })
   }
 
+  const stateWithFirstOpenedDaysAgo = (days: number) => {
+    return {
+      firstOpened: Date.now() - interval(`${days} days`),
+    }
+  }
+
+  const mountWithState = (query: Partial<SpecsListBannersFragment>, state?: Partial<AllowedState>, props?: object) => {
+    cy.mountFragment(SpecsListBannersFragmentDoc, {
+      onResult: (result) => {
+        assignIn(result, query)
+        set(result, 'currentProject.savedState', state)
+      },
+      render: (gql) => <SpecsListBanners gql={gql} {...props} />,
+    })
+  }
+
+  const validateSmartNotificationBehaviors = (bannerId: BannerId, bannerTestId: string, gql: Partial<SpecsListBannersFragment>) => {
+    context('banner conditions are met and when cypress use >= 4 days', () => {
+      it('should render when not previously-dismissed', () => {
+        mountWithState(gql, stateWithFirstOpenedDaysAgo(4))
+        cy.get(`[data-cy="${bannerTestId}"]`).should('be.visible')
+      })
+
+      it('should be preempted by spec not found banner', () => {
+        mountWithState(gql, stateWithFirstOpenedDaysAgo(4), { isSpecNotFound: true })
+        cy.get(`[data-cy="${bannerTestId}"]`).should('not.exist')
+      })
+
+      it('should be preempted by offline warning banner', () => {
+        mountWithState(gql, stateWithFirstOpenedDaysAgo(4), { isOffline: true })
+        cy.get(`[data-cy="${bannerTestId}"]`).should('not.exist')
+      })
+
+      it('should be preempted by fetch error banner', () => {
+        mountWithState(gql, stateWithFirstOpenedDaysAgo(4), { isFetchError: true })
+        cy.get(`[data-cy="${bannerTestId}"]`).should('not.exist')
+      })
+
+      it('should be preempted by project not found banner', () => {
+        mountWithState(gql, stateWithFirstOpenedDaysAgo(4), { isProjectNotFound: true })
+        cy.get(`[data-cy="${bannerTestId}"]`).should('not.exist')
+      })
+
+      it('should be preempted by request access banner', () => {
+        mountWithState(gql, stateWithFirstOpenedDaysAgo(4), { isProjectUnauthorized: true })
+        cy.get(`[data-cy="${bannerTestId}"]`).should('not.exist')
+      })
+
+      it('should not render when previously-dismissed', () => {
+        mountWithState(gql, {
+          ...stateWithFirstOpenedDaysAgo(4),
+          banners: {
+            [bannerId]: {
+              dismissed: Date.now(),
+            },
+          },
+        })
+
+        cy.get(`[data-cy="${bannerTestId}"]`).should('not.exist')
+      })
+    })
+
+    it('should not render when using cypress < 4 days', () => {
```

Comment:
Wonder if we should we wrap this in its own `context`, or move it to the top. It ends up jumping to the top in the UI anyway, which was just a little confusing when reviewing the tests.

![Screen Shot 2022-08-16 at 11 50 46 AM](https://user-images.githubusercontent.com/8340719/184923470-5483bc5e-d63c-472c-a0cb-595fdecbf05f.png)

Reply:
Moved tests to better align with execution order